src/pyhf/optimize/opt_minuit.py

```diff
@@ -27,11 +27,11 @@ def f(pars):
             'error_p{}'.format(i): (b[1] - b[0]) / float(self.steps)
             for i, b in enumerate(init_bounds)
         }
-        if constrained_mu is not None:
-            constraints = {'fix_p{}'.format(pdf.config.poi_index): True}
-            initvals['p{}'.format(pdf.config.poi_index)] = constrained_mu
-        else:
-            constraints = {}
+        fixed_vals = fixed_vals or []
```

Comment:
`fixed_vals=[]` instead of `None` by default. Then no need for this line.

Reply:
it's generally discouraged to have mutable containers in default args

https://nikos7am.com/posts/mutable-default-arguments/